addons/mail/static/src/models/attachment/attachment.js

```diff
@@ -109,7 +109,11 @@ function factory(dependencies) {
                 return `/web/image/${this.id}?signature=${this.checksum}`;
             }
             if (this.isPdf) {
-                return `/web/static/lib/pdfjs/web/viewer.html?file=/web/content/${this.id}`;
+                if (!this.accessToken && this.originThread && this.originThread.model === 'mail.channel') {
+                    return `/web/static/lib/pdfjs/web/viewer.html?file=/mail/channel/${this.originThread.id}/attachment/${this.id}`;
+                }
+                const accessToken = this.accessToken ? `?access_token%3D${this.accessToken}` : '';
+                return `/web/static/lib/pdfjs/web/viewer.html?file=/web/content/${this.id}${accessToken}`;
```

Comment:
Ideally the `/web/static/lib/pdfjs/web/viewer.html?file=` part shouldn't be duplicated, it's the same logic in both cases, only the file url changes.

Reply:
Done.